### ISSUE TITLE
Water Meter Entity Associating With Unit of Measurement

### ISSUE THREAD

**Issue by calanthe23** [cpyarger/Home-Assistant-Addons#19]
The signal from my water meter is being received, but it is not associated with the unit of measurement that I have chosen in the configuration.  Because of this, I cannot select my water usage within Home Assistant's energy monitoring interface.  Gas works great, however.

Data from the logs:
```
Allocating 15 zero-copy buffers
13:29:01.392523 decode.go:45: CenterFreq: 912380000
13:29:01.393266 decode.go:46: SampleRate: 2359296
13:29:01.393296 decode.go:47: DataRate: 32768
13:29:01.393308 decode.go:48: ChipLength: 72
13:29:01.393320 decode.go:49: PreambleSymbols: 32
13:29:01.393332 decode.go:50: PreambleLength: 4608
13:29:01.393343 decode.go:51: PacketSymbols: 736
13:29:01.393354 decode.go:52: PacketLength: 105984
13:29:01.393375 decode.go:59: Protocols: scm,scm+,idm,r900
13:29:01.393464 decode.go:60: Preambles: 111110010101001100000,0001011010100011,01010101010101010001011010100011,00000000000000001110010101100100
13:29:01.393508 main.go:124: GainCount: 29
RTLAMR JSON Output
{"Time":"2022-12-17T13:29:04.344522386-05:00","Offset":0,"Length":0,"Type":"SCM+","Message":{"FrameSync":5795,"ProtocolID":30,"EndpointType":171,"EndpointID":71962620,"Consumption":17265,"Tamper":18688,"PacketCRC":25782}}
JSON Output to HA REST API
{"state":"17265","attributes":{"unique_id":"71962620-sdrmr"}}
200
13:29:21.395174 main.go:341: Time Limit Reached: 20.001547545s
Signal caught, exiting!
Signal caught, exiting!
```

Configuration:
```
debug: true
rtltcpdebug: false
msgType: all
ids: 13629571,71962620
duration: 20
pause_time: 120
gas_unit_of_measurement: ft³
electric_unit_of_measurement: kWh
water_unit_of_measurement: gal
water_use_tenths: false
scm_plus_gas_divisor: 1
```

Entity properties:
![image](https://user-images.githubusercontent.com/17151197/208256487-429c0d81-3b94-475e-9595-e25a7dbbd2a3.png)

**Comment by pasyn** [comment]
Type 171 will need to be added to the code as a water meter. Right now it is not identified as water so it falls back to a generic meter with no units. I'll try to push the changes sometime today.

**Comment by pasyn** [comment]
Also, what is the model of the water meter and the units it is reporting? Some water meters report in tenths.

**Comment by calanthe23** [comment]
I appear to have an Itron EWQ-100WD - https://fccid.io/EWQ100WD

![image](https://user-images.githubusercontent.com/17151197/208779083-20858a35-200b-43e0-96e4-dd8cb7197b07.png)

Based on the data that I receive from it, I have only ever recorded whole numbers and not tenths.

**Comment by pasyn** [comment]
Thanks for the info. I plan on making a database of meters and what units they report to aid in setup.

I have pushed a fix. Update your addon and let me know how it goes.

**Comment by calanthe23** [comment]
@pasyn Thank you for pushing a fix so quickly!  I updated and have let it soak for a day.  Unfortunately, the unit of measure is still not being associated with the water meter entity in HA.  However, I can see where it is now being represented within the logs.  When I compare the placement of the attribute for the unit of measure for the gas reading to the water reading, the attribute is in a different position in the JSON stanza.  Is that correct?  My gas entity does work correctly and properly associates with the unit of measure.  See below as an example:

```
Allocating 15 zero-copy buffers
05:59:03.076929 decode.go:45: CenterFreq: 912380000
05:59:03.077969 decode.go:46: SampleRate: 2359296
05:59:03.077997 decode.go:47: DataRate: 32768
05:59:03.078012 decode.go:48: ChipLength: 72
05:59:03.078027 decode.go:49: PreambleSymbols: 32
05:59:03.078041 decode.go:50: PreambleLength: 4608
05:59:03.078055 decode.go:51: PacketSymbols: 736
05:59:03.078069 decode.go:52: PacketLength: 105984
05:59:03.078092 decode.go:59: Protocols: scm,scm+,idm,r900
05:59:03.078144 decode.go:60: Preambles: 0001011010100011,01010101010101010001011010100011,00000000000000001110010101100100,111110010101001100000
05:59:03.078177 main.go:124: GainCount: 29
RTLAMR JSON Output
{"Time":"2022-12-22T05:59:17.190810163-05:00","Offset":0,"Length":0,"Type":"SCM","Message":{"ID":13729571,"Type":12,"TamperPhy":3,"TamperEnc":0,"Consumption":270064,"ChecksumVal":28733}}
JSON Output to HA REST API
{"state":"270064","attributes":{"unique_id":"13729571-sdrmr","state_class":"total_increasing","device_class":"gas","unit_of_measurement":"ft³"}}
200
RTLAMR JSON Output
{"Time":"2022-12-22T05:59:19.656183477-05:00","Offset":0,"Length":0,"Type":"SCM+","Message":{"FrameSync":5795,"ProtocolID":30,"EndpointType":171,"EndpointID":71962620,"Consumption":17315,"Tamper":18688,"PacketCRC":28482}}
JSON Output to HA REST API
{"state":"17315","attributes":{"unique_id":"71962620-sdrmr"},"unit_of_measurement":"gal"}
200
05:59:23.083378 main.go:341: Time Limit Reached: 20.005108038s
Signal caught, exiting!
Signal caught, exiting!
```

**Comment by pasyn** [comment]
Looks like the bracket is in the wrong place in the json output. I'll push another update in a moment to fix. Afterwards, please confirm that the unit_of_measure attribute is shown in DeveloperTools>States.

**Comment by calanthe23** [comment]
YES!  That did it, now it has the attribute associated.  I validated in Developer Tools > States.  And, the log output looks like: `{"state":"17332","attributes":{"unique_id":"71962620-sdrmr","unit_of_measurement":"gal"}}`

Somewhat related: is it possible to also associate a state_class and device_class with this water meter?  If you look above at the logs I posted previously, for the gas meter, it has those attributes.  It appears that the energy dashboard in Home Assistant requires these to be set in order to represent water use on the dashboard.

Based on this documentation: https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics

I believe that they should be as follows:
`"state_class":"total_increasing","device_class":"water"`

With that change, the JSON would look like:
`{"state":"17332","attributes":{"unique_id":"71962620-sdrmr","state_class":"total_increasing","device_class":"water","unit_of_measurement":"gal"}}`

And I do believe that this would allow for water measurement to appear on the dashboard.

Thank you, again!
